frontend/apps/reader/modules/readerwikipedia.lua

```diff
@@ -385,7 +385,7 @@ end
 
 function ReaderWikipedia:lookupWikipedia(word, box, get_fullpage, forced_lang)
     if not NetworkMgr:isOnline() then
-        NetworkMgr:beforeWifiAction(function() ReaderWikipedia:lookupWikipedia(word, box, get_fullpage, forced_lang) end)
```

Comment:
May be it would work better with using `self` :  `function() self:lookupWikipedia(...) end)` (the instance, not the class).
Also beware that the Wikipedia might be using Trapper and coroutines - dunno how that would interact with your recursions.

Reply:
Huh. That works, thanks ;).